### ISSUE TITLE
question about training of confidence map

### ISSUE THREAD

**Issue by XiaotaoGuo** [wvangansbeke/Sparse-Depth-Completion#10]
Hi,

I noticed that your loss calculation has nothing to do with the confidence map of either local net or global net. So I'm curious that how do you train with those two maps? Since I'm trying to using your algorithm to train on my dataset but the confidence map looks weird.

**Comment by wvangansbeke** [comment]
Hi @XiaotaoGuo,

Indeed. I don't use any loss on the confidence maps. Due to the Softmax layer, the maps are between 0 and 1 and also complementary. We finally multiply the depth maps with their corresponding confidence map. So you don't need to set an additional loss on these maps, since the appearance and interpretation of the maps are determined by the architecture. You can see these maps as adaptive valves to decide what to take from the global branch or local 

For the final Github implementation, I changed some parts of the architecture. I added fusion at multiple layers. So more features from the global branch are combined with the features from the local branch. I found this to be more faster during training. Initially, I only did a multiplication at the beginning of the local branch with the guidance map (between 0 - 1) of the global branch (as described in the paper). This could be interpreted as noise reduction in the LiDAR frame. The advantage of this technique is that it results in interpretable confidence maps (which makes sense). This also makes it easier to find bugs. Now, the confidence maps are less interpretable, but training is faster and more stable.


Best,
Wouter

**Comment by XiaotaoGuo** [comment]
Thanks for your explanation
